CorsixTH/Lua/entities/machine.lua

```diff
@@ -237,7 +237,8 @@ function Machine:createHandymanActions(handyman)
   end
 
   local meander_loop_callback = --[[persistable:handyman_meander_repair_loop_callback]] function()
-    if not self.user then
+    -- Wait until the machine is not in use and not about to be used
+    if not self.user and (not self:getRoom():getPatient() or self:getRoom():getPatient().isLeaving) then
```

Comment:
You call the two functions twice here. You'd probably be better making a `local patient = self:getRoom():getPatient()` then in the if:
` if not self.user and (not patient or patient.isLeaving) then`

Reply:
Thanks. Not sure why I didn't do that sooner